templates/foreman.service-overrides.erb

```diff
@@ -3,7 +3,9 @@ User=<%= scope['foreman::user'] %>
 Environment=FOREMAN_ENV=<%= scope['foreman::rails_env'] %>
 Environment=FOREMAN_HOME=<%= scope['foreman::app_root'] %>
 Environment=FOREMAN_BIND=<%= scope['foreman::foreman_service_bind'] %>
+<%- if scope['foreman::foreman_service_port'] %>
 Environment=FOREMAN_PORT=<%= scope['foreman::foreman_service_port'] %>
+<%- end %>
```

Comment:
If I recall correctly, we have `FOREMAN_PORT=3000` straight in the service unit file and this file contains just overrides. So if `$foreman_service_port` is undefined, the default will be used. How does puma handle having bind set to a path and port to a number?

Reply:
My reading of https://github.com/puma/puma/blob/v4.3.3/lib/puma/binder.rb#L89-L175 says `puma` doesn't care about a port when it's doing `unix` sockets. However I remember @ehelms having fun with `rails server` yesterday, so it's probably good to test.

Reply:
But this will be interesting, see https://github.com/theforeman/foreman/pull/7972#discussion_r485439002

Reply:
If you look at the related PR (https://github.com/theforeman/foreman/pull/7972) the idea is to drop the default setting of those environment variables in the systemd file and rely on defaults in the Puma config or defaults from Puma itself if a user deploys this on their own without customizing through the installer.

Reply:
I think we can set it unconditionally since `foreman.service` also sets it hardcoded. If you don't want a value, you must clear it.

Reply:
Would that mean you end up with:

```
Environment=FOREMAN_PORT=undef
```

Reply:
I shouldn't do such superficial code reviews. Didn't see it could be undef.

However, thinking more about it (and spending some time on this today) I have some more thoughts.

It kind of depends on how you want to deal with the Foreman side. One option is to change things around. Rather than exposing a `FOREMAN_BIND` (which is right now really `FOREMAN_HOST`) and a `FOREMAN_PORT`, perhaps we want to expose `FOREMAN_BIND` as what Puma accepts. So `tcp://127.0.0.1:3000` or `unix:///run/foreman.sock`.

So I'm suggesting to consider getting rid of `FOREMAN_PORT` altogether.

Reply:
I think that's OK. This puts more onus on the user who wants to customize and configure to know the exact right combination rather than specifying an IP and Port which feels a bit more natural from a user input perspective. However, I really only expect container based users to really get in here and change these values.

Reply:
On the other hand, it allows users to tweak the binding. Puma can also bind on HTTPS. This allows users to run Foreman without Apache over HTTPS.

And containers don't use systemd so that's mostly irrelevant.